pubsub/pulsar/metadata.go

```diff
@@ -13,13 +13,18 @@ limitations under the License.
 
 package pulsar
 
+import "time"
+
 type pulsarMetadata struct {
-	Host            string `json:"host"`
-	ConsumerID      string `json:"consumerID"`
-	EnableTLS       bool   `json:"enableTLS"`
-	DisableBatching bool   `json:"disableBatching"`
-	Tenant          string `json:"tenant"`
-	Namespace       string `json:"namespace"`
-	Persistent      bool   `json:"persistent"`
-	Token           string `json:"token"`
+	Host                    string        `json:"host"`
+	ConsumerID              string        `json:"consumerID"`
+	EnableTLS               bool          `json:"enableTLS"`
+	DisableBatching         bool          `json:"disableBatching"`
+	Tenant                  string        `json:"tenant"`
+	Namespace               string        `json:"namespace"`
+	Persistent              bool          `json:"persistent"`
+	Token                   string        `json:"token"`
+	BatchingMaxPublishDelay time.Duration `json:"batchingMaxPublishDelay"`
```

Comment:
Let's add the batching metadata right next to disable batching (since they are inter related). Same ordering for constants in `pulsar.go`.